lib/services/page-service.ts

```diff
@@ -0,0 +1,152 @@
+                    const ts = packageJson.indexOf("nativescript-dev-typescript") > -1;
+
+                    switch (true) {
+                        case ng :
+                            resolve("Angular & TypeScript");
+                            break;
+                        case ts :
+                            resolve("TypeScript");
+                            break;
+                        default:
+                            resolve("JavaScript");
+
+                    }
+                })
+                .catch((promiseError: any) => {
+                   reject(promiseError);
+                });
+        });
+    }
+
+    addPage(pageName: string, appPath: string, pageTemplate: any) {
+        return new Promise((resolve, reject) => {
+            const displayName = pageTemplate.displayName.toLowerCase();
+
+            util.pageExists(appPath, pageName)
+                .then((pageExists: boolean) => {
+                    if (!pageExists) {
+                        return this.downloadPage(displayName, pageTemplate.flavor);
+                    } else {
+                        resolve(`Page with the name: ${pageName} already exists`);
+                    }
+                })
+                .then((downloadPath: any) => {
+                // TODO Add ejs template rendering here
+                resolve("success");
+                })
+                .catch((promiseError) => {
+                    reject(promiseError);
+                });
+        });
+    }
+
+    private getAppPath(appPath: string) {
+        return new Promise((resolve, reject) => {
+            if (!util.path.isAbsolute(appPath)) {
+                reject("Path must be absolute");
+            }
+
+            appPath = util.path.normalize(appPath);
+
+            util.fs.lstat(appPath, (statErr: any, stats: any) => {
+                if (statErr) {
+                    reject(statErr);
+                }
+
+                if (!stats.isDirectory()) {
+                    reject("Not a valid app folder!");
+                } else {
+                    resolve(appPath);
+                }
+            });
+
+        });
+    }
+
+    private readAppsPackageJson(appPath: string) {
+        return new Promise((resolve, reject) => {
+            util.fs.readdir(appPath, (readErr: any, files: any) => {
+                if (readErr) {
+                    reject(readErr);
+                }
+
+                if (files.indexOf("package.json") > -1) {
+                   const packageJsonPath = util.path.join(appPath, "package.json");
+
+                   util.fs.readFile(packageJsonPath, "utf8", (fileErr: any, data: any) => {
+                       if (fileErr) {
+                           reject(readErr);
+                       }
+                       resolve(data);
+                   });
+                } else {
+                    reject("Missing package.json file in path " + appPath);
```

Comment:
Better make the rejection reason an instanceof Error.

Reply:
Done